subsys/bluetooth/host/hci_core.c

```diff
@@ -3856,7 +3856,7 @@ static void le_pkey_complete(struct net_buf *buf)
 		atomic_set_bit(bt_dev.flags, BT_DEV_HAS_PUB_KEY);
 	}
 
-	for (cb = pub_key_cb; cb; cb = cb->_next) {
+	for (cb = pub_key_cb; cb && cb->func; cb = cb->_next) {
```

Comment:
When does the cb->func become NULL, and should the for loop really stop because of this?
Maybe that should just be in if inside the for loop to skip calling the func in that case?